WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -81,12 +81,26 @@ def edit
   end
 
   def destroy
-    @registration = Registration.find(params[:id])
-    @registration.destroy
-    mailer = RegistrationsMailer.notify_registrant_of_deleted_registration(@registration)
-    mailer.deliver_now
-    flash[:success] = "Deleted registration and emailed #{mailer.to.join(" ")}"
-    redirect_to competition_edit_registrations_path(@registration.competition)
+    if !current_user
+      return
+    end
+
+    @competition = competition_from_params
+    is_user = params.has_key?(:user_is_deleting_theirself) || false
```

Comment:
Also, could we call the variable `user_is_deleting_theirself` rather than `is_user`? I know it's more verbose, but I believe it's more readable.